static/js/components/CourseCarousel.js

```diff
@@ -0,0 +1,128 @@
+// @flow
+/* global SETTINGS:false */
+import React, { useState } from "react"
+import Carousel from "nuka-carousel"
+import Dotdotdot from "react-dotdotdot"
+import R from "ramda"
+
+import Card from "./Card"
+
+import { courseAvailability, maxPrice } from "../lib/courses"
+import { embedlyThumbnail } from "../lib/url"
+
+import type { Course } from "../flow/discussionTypes"
+
+type Props = {|
+  title: string,
+  courses: Array<Course>,
+  setShowCourseDrawer: Function
+|}
+
+const CAROUSEL_PAGE_SIZE = 3
+
+const CAROUSEL_IMG_HEIGHT = 130
+const CAROUSEL_IMG_WIDTH = 306
+
+type CardProps = {|
+  course: Course,
+  setShowCourseDrawer: Function
+|}
+
+export const CarouselCourseCard = ({
```

Comment:
What do you think of putting `CarouselCourseCard` in a separate file?

Reply:
I left it in the same file because I thought it was unlikely that we'd want to use it as a separate component, and I wanted to separate it out from the `CourseCarousel` component just to keep things easy to read.

Reply:
sounds good